[SPARK-49464] Add documentations (#113)
apache/spark-kubernetes-operator: docs/metrics_logging.md

<!--
Licensed to the Apache Software Foundation (ASF) under one
or more contributor license agreements.  See the NOTICE file
distributed with this work for additional information
regarding copyright ownership.  The ASF licenses this file
to you under the Apache License, Version 2.0 (the
"License"); you may not use this file except in compliance
with the License.  You may obtain a copy of the License at

  http://www.apache.org/licenses/LICENSE-2.0

Unless required by applicable law or agreed to in writing,
software distributed under the License is distributed on an
"AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
KIND, either express or implied.  See the License for the
specific language governing permissions and limitations
under the License.
-->

> # Metrics

Reviewer: Why do we have a new file for this? I'd recommend to include this content into `configuration.md`.

Spark operator,
following [Apache Spark](https://spark.apache.org/docs/latest/monitoring.html#metrics),
has a configurable metrics system based on
the [Dropwizard Metrics Library](https://metrics.dropwizard.io/4.2.25/). Note that Spark Operator
does not have Spark UI, MetricsServlet
and PrometheusServlet from org.apache.spark.metrics.sink package are not supported. If you are

> interested in Prometheus metrics exporting, please take a look at below section `Forward Metrics to Prometheus`

Reviewer 2: Add hyper link?

| kubernetes.client.http.response.3xx                       | Meter      | Tracking the rates of HTTP Code 3xx responses (redirection) received from the Kubernetes API Server per response code.   |
| kubernetes.client.http.response.4xx                       | Meter      | Tracking the rates of HTTP Code 4xx responses (client error) received from the Kubernetes API Server per response code.  |
| kubernetes.client.http.response.5xx                       | Meter      | Tracking the rates of HTTP Code 5xx responses (server error) received from the Kubernetes API Server per response code.  |
| kubernetes.client.<ResourceName>.<Method>                 | Meter      | Tracking the rates of HTTP request for a combination of one Kubernetes resource and one http method                      |
| kubernetes.client.<NamespaceName>.<ResourceName>.<Method> | Meter      | Tracking the rates of HTTP request for a combination of one namespace-scoped Kubernetes resource and one http method     |

## Forward Metrics to Prometheus

> In this section, we will show you how to forward spark operator metrics

Reviewer 2: ```suggestion
In this section, we will show you how to forward Spark Operator metrics
```

to [Prometheus](https://prometheus.io).

> * Modify the
>   build-tools/helm/spark-kubernetes-operator/values.yaml file' s metrics properties section:

Reviewer 2: ```suggestion
* Modify the metrics properties section in the file
  `build-tools/helm/spark-kubernetes-operator/values.yaml`:
```

```properties
metrics.properties:|+
  spark.metrics.conf.operator.sink.prometheus.class=org.apache.spark.kubernetes.operator.metrics.
sink.PrometheusPullModelSink
```

> * Install the Spark Operator

Reviewer 2: ```suggestion
* Install Spark Operator
```
